packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -223,6 +208,27 @@ const isLastBlockImageOrCode = (editor) => {
   return false;
 };
 
+/**
+ * Function that finds the corresponding block key in the blocks object based on the targetType and targetLevel (if present) provided.
+ * @param {object} blocks - blocks object
+ * @param {string} targetType - type of the target block
+ * @param {number} targetLevel - level of the target block
+ * @returns string|null - block key or null if no matching block is found
+ */
+const findBlockKeyByTypeAndLevel = (blocks, targetType, targetLevel = null) => {
+  const blockKey = Object.keys(blocks).find((key) => {
+    const block = blocks[key];
+
+    return (
+      block.value &&
+      block.value.type === targetType &&
+      (targetLevel === null || block.value.level === targetLevel)
+    );
+  });
+
+  return blockKey || null; // Return null if no matching block is found
+};
```

Comment:
const getAnchorBlockKey = (blocks, { type, level = null }) => {
  const anchorBlockKey = Object.keys(blocks).find((key) => {
    const block = blocks[key];

    return block.value.type === type && (level === null || block.value.level === level);
  });

  return anchorBlockKey;
};

I would like to suggest different name, also no need to return explicit null and block.value check